### DISCUSSION TITLE
Bug: Extend Config Inside Yarn Monorepo Does Not Work

### DISCUSSION THREAD

**Discussion by sashokbg** [eslint/eslint#18150] [Q&A] [answered]
### Environment

Node version:  v18.15.0
yarn version: 3.6.0

Local ESLint version: 8.57.0
Global ESLint version: N/A
Operating System: 6.7.4-arch1-1

### What parser are you using?

@typescript-eslint/parser

### What did you do?

In a yarn monorepo I have one package called @corporate/eslint containing the following eslint config:

```
module.exports = {
  'root': true,
  'env': {
    'browser': true,
    'es2021': true,
    'node': true,
    'jest': true,
  },
  'parser': '@typescript-eslint/parser',
  'extends': [
    'plugin:react/recommended',
    'plugin:prettier/recommended',
    'plugin:@typescript-eslint/recommended',
  ],
  'ignorePatterns': [
    '**/*.scss',
    '**/build/',
    '**/*.config.js',
    '**/dist/**',
    '**/jquery.qtip.min.js',
    '.eslintrc.js'],
  'parserOptions': {
    'ecmaVersion': 'latest',
    'sourceType': 'module',
  },
  'plugins': [
    'react',
    'react-hooks',
    '@typescript-eslint/eslint-plugin',
  ],
  'rules': {
    '@typescript-eslint/no-unused-vars': 1,
    '@typescript-eslint/ban-ts-comment': 1,
    '@typescript-eslint/no-explicit-any': 1,
    'react/react-in-jsx-scope': 1,
  },
  'settings': {
    'react': {
      'version': 'detect',
    },
  },
}
```

In a sibling package I am trying to extend this config using:

```
{
  "extends": [
    "@corporate/linter"
  ]
}
```

### What did you expect to happen?

When running lerna run lint:fix --scope @corporate/sibling
I expect the config to be loaded from the @corporate/linter module

### What actually happened?

Instead I get an error:

```
Oops! Something went wrong! :(

ESLint: 8.57.0

ESLint couldn't find the config "@corporate/linter" to extend from. Please check that the name of the config is correct.

The config "@corporate/linter" was referenced from the config file in "/home/alexander/projects/app/packages/corporate-cibling/.eslintrc.json".

If you still have problems, please stop by https://eslint.org/chat/help to chat with the team.
```

### Link to Minimal Reproducible Example

https://github.com/sashokbg/eslint-sharable-config

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I tried activating hoisting both on workspace and package level, but it didn't work.

What worked however is to reference the node_modules directory like so:

```
{
  "extends": [
    "../../../node_modules/@corporate/linter"
  ]
}
```

In the example repo run "lerna run lint" to reproduce

**Comment by Rec0iL99** [comment, marked as answer]
Hi @sashokbg, this is happening because your config is not following the [naming convention](https://eslint.org/docs/latest/extend/shareable-configs#creating-a-shareable-config) ESLint has in place for shareable configs. Config name should be prefixed with `eslint-config`, so your config name should be `@corporate/eslint-config-lint`.

**Reply by sashokbg** [reply]
silly me .. sorry for posting an RTFM question ! :)

**Reply by aryanprince** [reply]
> Snippet from docs: While you can **name the package in any way that you’d like**, we recommend using one of the following conventions to make your package easier to identify [(source)](https://eslint.org/docs/latest/extend/shareable-configs#creating-a-shareable-config:~:text=While%20you%20can%20name%20the%20package%20in%20any%20way%20that%20you%E2%80%99d%20like%2C%20we%20recommend%20using%20one%20of%20the%20following%20conventions%20to%20make%20your%20package%20easier%20to%20identify)

The documentation states that you can **name your shared ESLint package however you like**. However, how can other apps in a monorepo detect our shared ESLint package if we **don’t** follow the recommended naming conventions?

In my case, our internal config packages use the following pattern: `@org/config-*`. For example, we have `@org/config-prettier`, `@org/config-tailwind`, and `@org/config-typescript`. But `@org/config-eslint` wouldn’t work because it doesn’t follow the naming conventions mentioned [here](https://eslint.org/docs/latest/extend/shareable-configs#creating-a-shareable-config).

Since the documentation says it’s possible to name the package in any way, what are the alternatives? Are there other methods besides the one mentioned by @sashokbg where you manually put the path to your shared config from your `node_modules`?

```
{
  "extends": [
    "../../../node_modules/@org/config-eslint"
  ]
}
```

**Reply by fregante** [reply]
> The documentation states that you can name your shared ESLint package however you like

Correct, however the code suggests otherwise. It only started working for me when I named the package `@org/eslint-config` (ESLint 8.57). You can see this line:

https://github.com/eslint/eslintrc/blob/18d22b7c58546e0eafce78ceb9961271c7d0d4c7/lib/config-array-factory.js#L927-L930
